app/src/main/kotlin/com/wire/android/mapper/MessagePreviewContentMapper.kt

```diff
@@ -86,18 +86,34 @@ private fun MessagePreview.uiLastMessageContent(): UILastMessageContent {
                     userUIText,
                     UIText.PluralResource(
                         R.plurals.last_message_people_added,
-                        (content as WithUser.MembersAdded).count,
-                        (content as WithUser.MembersAdded).count,
-                    )
-                )
-                is WithUser.MembersRemoved -> UILastMessageContent.SenderWithMessage(
-                    userUIText,
-                    UIText.PluralResource(
-                        R.plurals.last_message_people_removed,
-                        (content as WithUser.MembersRemoved).count,
-                        (content as WithUser.MembersRemoved).count
+                        (content as WithUser.MembersAdded).userIdList.size,
+                        (content as WithUser.MembersAdded).userIdList.size,
                     )
                 )
+
+                is WithUser.MembersRemoved -> {
+                    val membersRemovedContent = (content as WithUser.MembersRemoved)
+                    val isSelfRemoved = membersRemovedContent.userIdList.contains(selfUserId)
+                    val isSelfLeft = isSelfMessage && isSelfRemoved
```

Comment:
Maybe we should also check if `selfId` is the only id on `userIdList`, I don't know if it's possible to remove self and others at the same time, but the message format doesn't prevent that 😄

Reply:
I'm not sure because `membersRemovedContent.isSelfUserRemoved` only verifies if the self user is on the list, but not if the `otherUserIdList` is empty, so `isSelfLeft` can be true even if there are more people besides self user on the removed list 🤔